kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -0,0 +1,177 @@
+"""``AbstractDataSet`` implementation to access Snowflake using Snowpark dataframes
+"""
+from copy import deepcopy
+from typing import Any, Dict, Union
+
+import pandas as pd
+import snowflake.snowpark as sp
+
+from kedro.io.core import AbstractDataSet, DataSetError
+
+
+# TODO: Update docstring after interface finalised
+# TODO: Add to docs example of using API to add dataset
+class SnowParkDataSet(
+    AbstractDataSet[pd.DataFrame, pd.DataFrame]
+):
+    """``SnowParkDataSet`` loads and saves Snowpark dataframes.
+
+    Example adding a catalog entry with
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+        data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+
+    .. code-block:: yaml
+
+        >>> weather:
+        >>>   type: snowflake.SnowParkDataSet
+        >>>   table_name: weather_data
+        >>>   warehouse: warehouse_warehouse
+        >>>   database: meteorology
+        >>>   schema: observations
+        >>>   credentials: db_credentials
+        >>>   load_args (WIP):
```

Comment:
```suggestion
        >>>   load_args (WIP):
```

You don't need in the example, but It would be good to get examples of doing this form of credentials, but also the `externalbroser` SSO approach :)